discovery-provider/src/api/v1/helpers.py

```diff
@@ -67,7 +67,11 @@ def add_playlist_added_timestamps(playlist):
     added_timestamps = []
     for track in playlist["playlist_contents"]["track_ids"]:
         added_timestamps.append(
-            {"track_id": encode_int_id(track["track"]), "timestamp": track["time"]}
+            {
+                "track_id": encode_int_id(track["track"]),
+                "timestamp": track["time"],
+                "metadata_timestamp": track.get("metadata_time", track["time"]),
```

Comment:
i think time always has to be there. if metadata_time is not there because these were playlists before adding metadata then it'd default to using the indexing timestamp instead.